src/internal/connector/exchange/service_iterators.go

```diff
@@ -421,7 +421,7 @@ func FetchMessageIDsFromDirectory(
 		deltaURL string
 	)
 
-	options, err := optionsForFolderMessagesDelta([]string{"id"})
+	options, err := optionsForFolderMessagesDelta([]string{"id", "isRead"})
```

Comment:
Just a reminder, you do not need to put `id` in the options as it is included in the buildOptions automatically.

```go
func buildOptions(options []string, optID optionIdentifier) ([]string, error) {
	var (
		allowedOptions  map[string]int
		returnedOptions = []string{"id"}
```